### PR TITLE
Expose symbolic files to command line interface, implement whence for symbolic seek

### PR DESCRIPTION
**Pull request by 251**
As requested in Slack:

* a new parameter `--file` to add symbolic files
* a very simple symbolic seek implementation (current `seek` lacks a `whence` parameter and fails with TypeError)

Edit: changed `symfile` to `file` for consistent naming

### CODE REVIEW

File: manticore/platforms/linux.py [hunk 1]
```diff
@@ -170,14 +170,25 @@ def tell(self):
...
+        assert isinstance(offset, (int, long))
+        assert isinstance(whence, (int, long))
+
+        # horribly simplified (omits mode checks, error handling)
+        if whence == os.SEEK_SET:
+            self.pos = offset
+        elif whence == os.SEEK_CUR:
+            self.pos += offset
+        elif whence == os.SEEK_END:
+            self.max_size += offset
```
Comment (japesinator) [inline manticore/platforms/linux.py]: I'm not sure this is correct behavior, I think this should just set `pos` to `offset` bytes from the end of the file and not affect `max_size`

File: manticore/platforms/linux.py [hunk 2]
```diff
@@ -170,14 +170,25 @@ def tell(self):
         '''
         return self.pos
 
-    def seek(self, pos):
+    def seek(self, offset, whence = os.SEEK_SET):
         '''
         Returns the read/write file offset
```
Comment (japesinator) [inline manticore/platforms/linux.py]: This docstring isn't correct anymore, we should probably update it

File: manticore/platforms/linux.py (hunk 1 above)
Comment (251) [inline manticore/platforms/linux.py]: What about a more complex version:

```Python
        new_position = 0
        if whence == os.SEEK_SET:
            new_position = offset
        elif whence == os.SEEK_CUR:
            new_position = self.pos + offset
        elif whence == os.SEEK_END:
            new_position = self.max_size + offset

        if new_position < 0:
            return -1

        if new_position < self.max_size:
            self.pos = new_position
        else:
            logger.warning("Symbolic seek: maximum file length reached")
            self.pos = self.max_size

        return self.pos
```

I'm not sure on how to handle the last case. I guess one can drop the last if/else. Technically there is no problem setting the position behind max_size. A read/write access should warn about it.

Comment (japesinator) [inline manticore/platforms/linux.py]: At least in the python repl, `seek` will let you move the cursor as far past the end of the file as you'd like, and I'm fine being consistent with that.

**Comment by japesinator** [comment]
This looks great, thanks!

**Pull request merged**